### DISCUSSION TITLE
Unable to read from chat database disk I/O error after full disk access

### DISCUSSION THREAD

**Discussion by ogwannabe** [ReagentX/imessage-exporter#176] [Q&A] [answered]
I am attempting to export messages from an iOS backup. When running the following command:
`sudo imessage-exporter -f html -p ~/<filepath>/Backup -c compatible`

I get an error of: 

```
Unable to launch: Unable to read from chat database: disk I/O error
Ensure full disk access is enabled for your terminal emulator in System Settings > Security and Privacy > Full Disk Access
```

I have already enabled full disk access to terminal, and enabled developer tools for Terminal.

What other permissions do I need to allow to run this?

**Comment by ReagentX** [comment, marked as answer]
In your command:

```zsh
sudo imessage-exporter -f html -p ~//Backup -c compatible
```

Are you sure the path to the backup is correct? `~//Backup` looks odd to me because it has two `//`.

Assuming it is, the software could not find your iMessage database. Is the backup unencrypted?

There should be a better error here that tell you what went wrong; I am tracking that in #177.

**Comment by plattsad** [comment]
Yeh - I think apple blocked all user level access to chat.db in sonoma.  I got the same error as the first poster even having given imessage-exporter full disk access and run it as both the intended user and as root:

`
sh-3.2#  /Users/pla/Downloads/imessage-exporter -c compatible -f html -m claud -p /Users/pla/Library/Messages/chat.db
Unable to launch: Unable to read from chat database: unable to open database file: /Users/pla/Library/Messages/chat.db
Ensure full disk access is enabled for your terminal emulator in System Settings > Security and Privacy > Full Disk Access

ls -alth  /Users/pla/Library/Messages/chat.db
-rw-r--r--  1 pla  staff    72M Dec 27 12:34 /Users/pla/Library/Messages/chat.db
`

**Reply by ReagentX** [reply]
`imessage-exporter` still definitely works. Did you grant your terminal emulator full disk access as well?

**Reply by plattsad** [reply]
That did the trick.  Sorry for doubting you.
